cp_diffdock_api.py

```diff
@@ -0,0 +1,261 @@
+                complex_name = f"{protein_entry.label}-{ligand_entry.label}"
+                docking_entries.append(PreppedComplex(complex_name, protein_entry, ligand_entry))
+
+        csv_file = work_dir / "bmaps_diffdock.csv"
+        csv_content = DiffDockApi.make_request_csv(docking_entries)
+        if csv_content:
+            try:
+                with open(csv_file, 'w') as f:
+                    f.write(csv_content)
+            except IOError as e:
+                raise DiffDockSetupException(f"Could not write diffdock csv file: {e.strerror}")
+        else:
+            log.info(f"Couldn't prepare diffdock csv. Entries: {docking_entries}")
+            raise DiffDockSetupException("No suitable complexes")
+
+        out_dir = work_dir / "results"
+        return PreppedRequest(csv_file, out_dir, docking_entries)
+
+    @staticmethod
+    def process_results(preppedRequest: PreppedRequest) -> List[DiffDockProtocol.Response]:
+        """
+        DiffDock produces sdf files for each docked pose, in directories by complex name.
+        This function gathers the resulting sdfs for each complex to be returned by the DiffDock API.
+        """
+        allComplexResults = []
+
+        for complexInfo in preppedRequest.complex_infos:
+            # Find the output directory for this complex
+            complexName, proteinUnit, ligandUnit = complexInfo
+            proteinName = proteinUnit.name
+            ligandName = ligandUnit.name
+            resultDir = preppedRequest.out_dir / complexName
+            if not os.path.isdir(resultDir):
+                noPosesError = "No poses generated."
+                if not proteinUnit.ok():
+                    noPosesError += f" protein prep failure: {proteinUnit.error}."
+                if not ligandUnit.ok():
+                    noPosesError += f" ligand prep failure: {ligandUnit.error}."
+                allComplexResults.append(DiffDockProtocol.Result(proteinName, ligandName, error=noPosesError))
+                continue
+
+            # Process the sdfs in the output directory
+            poses = []
+            for dir_entry in os.scandir(resultDir):
+                filepath = dir_entry.path
+                filename = os.path.basename(filepath)
+
+                # Regex to get rank and confidence
+                rankRegex = re.compile(r"rank(\d+)(_confidence(-?[0-9.]+))?\.sdf")
+                match = rankRegex.match(filename)
+
+                if match:
+                    rankStr = match.group(1)
+                    confidenceStr = match.group(3)
+                    rankInt = int(rankStr)
+                    confidenceFloat = -1
+                    if confidenceStr:
+                        confidenceFloat = float(confidenceStr)
+                    with open(filepath, 'r') as f:
+                        fileContent = f.read()
+                    poses.append(DiffDockProtocol.Pose(filename, fileContent, rankInt, confidenceFloat))
+
```

Comment:
Are other files expected in that directory? Or at least other .sdf if you use glob?
If not, something is anomalous and the regex fail should be logged.
If so, a comment could explain. Maybe an `else:` a comment, and a `pass`.